// eslint-plugin-yaml/lib/index.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.configs = exports.processors = void 0;
const js_yaml_1 = require("js-yaml");
const jshint_1 = require("jshint");
const fileContents = {};
function preprocess(text, fileName) {
    fileContents[fileName] = text;
    return [{ text: text, filename: fileName }];
}
function postprocess(messages, fileName) {
    let doc;
    try {
        doc = js_yaml_1.safeLoad(fileContents[fileName], {
            filename: fileName,
            json: false
        });
    }
    catch (e) {
        return [
            {
                ruleId: "invalid-yaml",
                severity: 2,
                message: e.message,
                source: e.mark.buffer,
                line: e.mark.line,
                column: e.mark.column
            }
        ];
    }
    const yaml_json = JSON.stringify(doc, null, 2);
    jshint_1.JSHINT(yaml_json);
    const data = jshint_1.JSHINT.data();
    const errors = (data && data.errors) || [];
    const linter_messages = errors
        .filter(function (e) {
        return !!e;
    }).map(function (error) {
        return {
            ruleId: "bad-yaml",
            severity: 2,
            message: error.reason,
            source: error.evidence,
            line: error.line,
            column: error.character
        };
    });
    return linter_messages;
}
exports.processors = {
    ".yml": {
        preprocess: preprocess,
        postprocess: postprocess
    },
    ".yaml": {
        preprocess: preprocess,
        postprocess: postprocess
    }
};
exports.configs = {
    recommended: {
        ignorePatterns: ["!.github"],
        overrides: [{
                plugins: ['yaml'],
                "files": ["*.yaml", "*.yml"],
            }]
    },
};
